apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/extract/s3/DataSetManifestFactory.java

```diff
@@ -0,0 +1,100 @@
+package gov.cms.bfd.pipeline.ccw.rif.extract.s3;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import javax.xml.XMLConstants;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.Schema;
+import javax.xml.validation.SchemaFactory;
+import org.xml.sax.SAXException;
+
+public class DataSetManifestFactory {
+  /** The {@link String} pipeline manifest XML Schema Definition (XSD). */
+  public static final String MANIFEST_XSD =
```

Comment:
Should this be externally loaded from an xml? Seems strange to have it in-code.

Reply:
possibly...however BFD to date has no support for URL based file services. For this case, the XSD is so limited it is probably not worth the effort to have an XSD resource, available to both testing (build) and prod pipeline execution. In fact I debated whether to even do the right thing of using an XSD when I could have simply hard-coded checking the value of the 'syntheticData' attribute with a simple 'if' statement and throw a JAXBException if it wasn't 'true' or 'false'.

Reply:
Can't we just load it from the classpath as a resource? Having this in the file system would allow us to pass it along to mitre and ccw if they ever wanted to validate the manifests that they generate, and we could add it to the synthetic automation as an extra layer of protection so that we don't blow up the pipeline with a bad manifest.

Reply:
loaded from file; however, it's not like we can actually publish this XSD since BFD is brain-dead for static file resources.